tests/test_optimize.py

```diff
@@ -671,3 +671,65 @@ def test_update_stepsize(self):
         eta2 = 0.1
         opt.update_stepsize(eta2)
         assert opt._stepsize == eta2
+
+    @pytest.mark.parametrize(
+        "opt",
+        [
+            GradientDescentOptimizer(stepsize),
+            MomentumOptimizer(stepsize, momentum=gamma),
+            NesterovMomentumOptimizer(stepsize, momentum=gamma),
+            AdagradOptimizer(stepsize),
+            RMSPropOptimizer(stepsize, decay=gamma),
+            AdamOptimizer(stepsize, beta1=gamma, beta2=delta),
+            RotosolveOptimizer(),
+        ],
+    )
+    class Test_over_opts:
+        def test_kwargs(self, opt, tol):
```

Comment:
I think there is an indentation issue here; this class/decorator should be at the module level (no indentation).

Actually, I'm a bit surprised this works! I didn't think pytest would find/test nested classes.

Reply:
So I should take it out of the `TestOptimizer` class?

Reply:
Think what @josh146 meant was to make the new class be another stand-alone class, just as `TestOptimizer`.

Reply:
Yep!

Reply:
will do.